generate_report.py

```diff
@@ -1,20 +1,139 @@
+  service = build(api_name, api_version, http=http)
+
+  return service
+
+def get_first_profile_id(service):
+  # Use the Analytics service object to get the first profile id.
+
+  # Get a list of all Google Analytics accounts for this user
+  accounts = service.management().accounts().list().execute()
+
+  if accounts.get('items'):
+    # Get the first Google Analytics account.
+    account = accounts.get('items')[0].get('id')
+
+    # Get a list of all the properties for the first account.
+    properties = service.management().webproperties().list(
+        accountId=account).execute()
 
+    if properties.get('items'):
+      # Get the first property id.
+      property = properties.get('items')[0].get('id')
+
+      # Get a list of all views (profiles) for the first property.
+      profiles = service.management().profiles().list(
+          accountId=account,
+          webPropertyId=property).execute()
+
+      if profiles.get('items'):
+        # return the first view (profile) id.
+        return profiles.get('items')[0].get('id')
+
+  return None
+
+
+def get_n_results(service, profile_id, n):
+  # Use the Analytics Service Object to query the Core Reporting API
+  # for the number of sessions within the past seven days.
+  ndate = str(n)+'daysAgo'
+  return service.data().ga().get(
+      ids='ga:' + profile_id,
+      start_date=ndate,
+      end_date='today',
+      metrics='ga:sessions').execute()
 
 @click.command()
-@click.option('--name', default='report.pdf', help='Name of output file.')
-def run(name):
-    click.echo('Hello %s!' % name)
+@click.option('--file', default='analytics.pdf', help='PDF File Name')
+def main(file):
+  # Define the auth scopes to request.
+  scope = ['https://www.googleapis.com/auth/analytics.readonly']
+
+  # Use the developer console and replace the values with your
+  # service account email and relative location of your key file.
+  if "GOOGLE_SERVICE_EMAIL" in os.environ:
+      service_account_email = os.environ["GOOGLE_SERVICE_EMAIL"]
+  else:
+      print("Please set the environment variable GOOGLE_SERVICE_EMAIL, which is your google api service account email")
+      sys.exit(1)
+
+  if "KEY_FILE_LOCATION" in os.environ:
+      key_file_location = os.environ["KEY_FILE_LOCATION"]
+
+  else:
+      print("Please set the environment variable KEY_FILE_LOCATION, which is the location to your .p12 client secrets file")
+      sys.exit(1)
+
+  # Authenticate and construct service.
+  service = get_service('analytics', 'v3', scope, key_file_location,
+    service_account_email)
+  profile = get_first_profile_id(service)
+  #xidN => n = # of days, and xid represents the data based off a set number of days(sessions)
+  xid7 = get_n_results(service, profile, 7)
+  xid14 = get_n_results(service, profile, 14)
+  xid21 = get_n_results(service, profile, 21)
+  xid28 = get_n_results(service, profile, 28)
+  xid35 = get_n_results(service, profile, 35)
+  print(int(xid28.get('rows')[0][0]))
```

Comment:
[DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), you can do this in a for loop.